src/java/fr/paris/lutece/plugins/document/business/DocumentDAO.java

```diff
@@ -671,216 +709,171 @@ public Collection<Integer> selectPrimaryKeysByFilter( DocumentFilter filter )
             for ( int nCategoryId : filter.getCategoriesId( ) )
             {
                 if ( nCategoryId > 0 )
                 {
-                    sbCategories.append( SQL_FILTER_CATEGORIES );
+                    filterCategories.add( SQL_FILTER_CATEGORIES );
                 }
                 else
                 {
-                    sbCategories.append( SQL_FILTER_CATEGORIES_NULL );
+                    filterCategories.add( SQL_FILTER_CATEGORIES_NULL );
                 }
-
-                if ( ( i + 1 ) < filter.getCategoriesId( ).length )
-                {
-                    sbCategories.append( SQL_FILTER_CATEGORIES_OR );
-                }
-
-                i++;
             }
-
-            sbCategories.append( SQL_FILTER_CATEGORIES_END );
-            sbWhere.append( ( sbWhere.length( ) != 0 ) ? SQL_FILTER_AND : StringUtils.EMPTY ).append( sbCategories.toString( ) );
+            wherePart.add( SQL_FILTER_CATEGORIES_BEGIN + StringUtils.join( filterCategories, SQL_FILTER_CATEGORIES_OR ) + SQL_FILTER_CATEGORIES_END );
         }
-
         if ( filter.containsIdsCriteria( ) )
         {
-            StringBuilder sbIds = new StringBuilder( SQL_FILTER_ID_BEGIN );
+            List<String> filterIds = new ArrayList<>( );
 
             for ( int i = 0; i < filter.getIds( ).length; i++ )
             {
-                sbIds.append( SQL_FILTER_ID );
-
-                if ( ( i + 1 ) < filter.getIds( ).length )
-                {
-                    sbIds.append( SQL_FILTER_ID_OR );
-                }
+                filterIds.add( SQL_FILTER_ID );
             }
-
-            sbIds.append( SQL_FILTER_ID_END );
-            sbWhere.append( ( sbWhere.length( ) != 0 ) ? SQL_FILTER_AND : StringUtils.EMPTY ).append( sbIds.toString( ) );
+            wherePart.add( SQL_FILTER_ID_BEGIN + StringUtils.join( filterIds, SQL_FILTER_ID_OR ) + SQL_FILTER_ID_END );
         }
 
         if ( BooleanUtils.isFalse( filter.isPublished( ) ) )
         {
-            sbWhere.append( ( sbWhere.length( ) != 0 ) ? SQL_FILTER_AND : StringUtils.EMPTY )
-                    .append( "a.id_document NOT IN (SELECT DISTINCT id_document FROM document_published) " );
+            wherePart.add( "a.id_document NOT IN (SELECT DISTINCT id_document FROM document_published) " );
         }
 
         if ( StringUtils.isNotBlank( filter.getDateMin( ) ) && StringUtils.isNotBlank( filter.getDateMax( ) ) )
         {
-            sbWhere.append( ( ( sbWhere.length( ) != 0 ) ? SQL_FILTER_AND : StringUtils.EMPTY ) ).append( "a.date_modification < " )
-                    .append( "'" + filter.getDateMax( ) + "'" ).append( SQL_FILTER_AND ).append( "a.date_modification > " )
-                    .append( "'" + filter.getDateMin( ) + "'" );
+            wherePart.add( "a.date_modification < "+ "'" + filter.getDateMax( ) + "'" + SQL_FILTER_AND + "a.date_modification > " + "'" + filter.getDateMin( ) + "'" );

```

Comment:
As part of this PR, would it be possible to modify this line: the value of the max date should not be positioned directly in the where clause, but added afterwards to avoid SQL injection security flaws, since it is possible that this value comes directly from a user input. (same thing for the other lines of this Class)